simexpal/base.py

```diff
@@ -1204,6 +1212,23 @@ def effective_thread_settings(self):
 			return s.thread_settings
 		return self.info.thread_settings
 
+	@property
+	def is_effective_exclusive(self):
```

Comment:
I don't quite understand the logic of this function. As I understand, this function shall return `True` if the experiment is setup to use computing resources exclusively. 

Now you iterate over all variants (why variants?) and then you check if a variant is exclusive: if it is, you continue, if it is not exclusive but no variant has been stored into `s` it'll also continue with the next element, and if the next element is also not exclusive you raise a `RuntimeError` with both the last and the current variant name (`s.name, variant.name`)?

I would be happy for some clarity on what you are doing here to understand how we can simplify this logic.

Reply:
The code itself is similar to the `effective_process_settings` and `effective_thread_settings` properties defined before it.

To my understanding, the logic is as follows:
- iterate over all variants, check their "settings" (be it process, threads or exclusiveness)
- if two variants have settings set, throw an Error
- if only one variant has settings set, return that
- if no variant has settings set, return the settings from the experiment

However, the code in `is_effective_exclusive` did indeed contain a bug and wasn't working as intended. I've fixed that and further adapted it: the logic is still similar to the above but the error is only thrown if the exclusiveness of two variants varies.

Furthermore, I think the code for `effective_process_settings` and `effective_thread_settings` doesn't account for the fact that the settings of two variants could be the same, in which case there would be no need to throw an error (like I do now)

Reply:
And the schema has been changed to allow `num_nodes` and `exclusive: true` to be set.